src/algorithms/optimized_shooting.jl

```diff
@@ -0,0 +1,116 @@
+export periodic_orbit, periodic_orbits, OptimizedShooting
+
+using LeastSquaresOptim: optimize, LevenbergMarquardt
+
+
+"""
+    OptimizedShooting(; kwargs...)
+
+A shooting method [Dednam2014](@cite) combined with Levenberg-Marquardt optimization 
+to find periodic orbits of continuous systems.
+
+## Keyword arguments
+- `Δt::Float64=1e-6`: explicit ODE solver time step. This should correspond to the time step used in the ODE solver specified in the `CoupledODEs` object.
+- `p::Int64=2`: `p*dimension(ds)` is the number of points in the residual `R`.
+- `optim_kwargs::NamedTuple=(x_tol=1e-10,)`: keyword arguments to pass to the optimizer. The optimizer used is the `optimize` from `LeastSquaresOptim.jl`. For details on the keywords see the respective package documentation.
+- `abstol::Float64=1e-3` : absolute tolerance for sum of squares `ssr` of the residual `R`. The method converged if `ssr <= abstol`.
+
+## Description
+
+Continuous dynamical system 
+
+```math
+\\frac{dx}{dt} = f(x, p, t)
+```
+
+can be rewritten in terms of the dimensionless time ``\\tau = t/T`` as
+
+```math
+\\frac{dx}{dt} = Tf(x, p, T\\tau)
+```
+where ``T`` is a period of some periodic orbit. The boundary conditions for the 
+periodic orbit now are ``x(\\tau = 0) = x(\\tau = 1)``. Dednam and Botha [Dednam2014](@cite) 
+suggest minimizing the residual ``R`` defined as 
+
+```math
+R = (x(1)-x(0), x(1+\\Delta \\tau)-x(\\Delta \\tau), ..., x(1+(p-1)\\Delta \\tau)-x((p-1)\\Delta \\tau))
+```
+
+ with respect to ``x(0)`` and ``T`` using Levenberg-Marquardt optimization.
+
+In our implementation keyword argument `p` corresponds to ``p`` in the residual ``R``. 
+The keyword argument `Δt` corresponds to ``\\Delta \\tau`` in the residual ``R``.
+
+## Important note
+
+For now we recommed using the `RKO65` ODE solver and setting the stepsize `dt` the 
+same as `Δt`. For example
+
+```
+alg = OptimizedShooting(Δt=1/(2^6), p=3)
+ds = CoupledODEs(dynamic_rule, state, params; diffeq = (alg=RKO65(), dt=alg.Δt))
+```
+
+Using other ODE solvers may lead to divergence.
+
+"""
+@kwdef struct OptimizedShooting <: PeriodicOrbitFinder
+    Δt ::Float64 = 1e-6
```

Comment:
Try to have consistent formatting. Don't have spaces between `::`. I will try to set up an automatic formatter for JuliaDynamics in the meantime.